src/snowflake/snowpark/dataframe_writer.py

```diff
@@ -256,6 +256,7 @@ def save_as_table(
             Dict[str, Union[str, Iterable[ColumnOrSqlExpr]]]
         ] = None,
         table_exists: Optional[bool] = None,
+        override_condition: Optional[ColumnOrSqlExpr] = None,
```

Comment:
I personally find the name `override` confusing -- override to me is more of a OO design concept where we replace the old behavior with a new one, but here we actually overwrite data.